test/metabase/query_processor/middleware/constraints_test.clj

```diff
@@ -51,3 +52,20 @@
            (add-default-userland-constraints
             {:constraints {:max-results 5, :max-results-bare-rows 10}
              :middleware  {:add-default-userland-constraints? true}})))))
+
+(deftest load-constraints-settings-from-config-test
+  (testing "loading user set constraint settings from config"
+    (mt/with-temporary-setting-values [max-results 10 max-results-bare-rows 5]
+      (is (= {:constraints {:max-results (setting/get :max-results)
+                            :max-results-bare-rows (setting/get :max-results-bare-rows)}}
```

Comment:
It's ok to assume that `with-temporary-setting-values` works correctly, so you don't need to turn around and fetch the values right after you set them:

```suggestion
      (is (= {:constraints {:max-results 10
                            :max-results-bare-rows 5}}
```

At this point, I'm not really sure what this is mean to test, `add-default-userland-constraints` only adds `:max-results` and `:max-results-bare-rows` if they're not already in the map, and only if the map has 

```clj
{:middleware {:add-default-userland-constraints? true}}
```

So this is basically just testing `(= x x)`

Reply:
I think `add-constraints-test` above is sufficient for testing your changes, I think you can actually just go ahead and take this test out since it's not adding anything